src/main/java/io/supertokens/webserver/api/multitenancy/CreateOrUpdateTenantOrGetTenantAPI.java

```diff
@@ -115,9 +79,16 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
                 throw new TenantOrAppNotFoundException(tenantIdentifier);
             }
             boolean shouldProtect = shouldProtectProtectedConfig(req);
-            JsonObject result = config.toJson(shouldProtect, getTenantStorage(req), CoreConfig.PROTECTED_CONFIGS);
+            JsonObject result = config.toJson(
+                    shouldProtect, getTenantStorage(req), CoreConfig.PROTECTED_CONFIGS,
+                    getVersionFromRequest(req).lesserThan(SemVer.v5_1));
             result.addProperty("status", "OK");
 
+            if (getVersionFromRequest(req).lesserThan(SemVer.v5_1)) {
+                result.get("thirdParty").getAsJsonObject().remove("useThirdPartyProvidersFromStaticConfigIfEmpty");
+                result.remove("useFirstFactorsFromStaticConfigIfEmpty");
+            }
+
             if (getVersionFromRequest(req).lesserThan(SemVer.v5_0)) {
                 result.remove("firstFactors");
```

Comment:
you should make the toJson function only do this logic? Otherwise it is easy to miss doing this

Reply:
Done